cmd/yggd/exec.go

```diff
@@ -5,135 +5,80 @@ import (
 	"io"
 	"os"
 	"os/exec"
-	"path/filepath"
-	"strings"
-	"time"
 
 	"git.sr.ht/~spc/go-log"
-	"github.com/redhatinsights/yggdrasil"
 )
 
-func startProcess(file string, env []string, delay time.Duration, died chan int) {
+// startProcess executes file, setting up the environment using the provided env
+// values. If started is not nil, it is invoked on a goroutine after the process
+// has been started.
+func startProcess(
+	file string,
+	args []string,
+	env []string,
+	started func(pid int, stdout io.ReadCloser, stderr io.ReadCloser),
```

Comment:
Good point. I spent some time researching conventions around naming function parameters and couldn't find any definitive sources. There's lots of opinion on the Internet (no surprise). I try to follow the Go standards as much as possible, and in the absence of documentation by the Go project (as is the case here), I look at the Go standard library for inspiration.

* The [net/http](https://pkg.go.dev/net/http) package declares a function called [HandleFunc](https://pkg.go.dev/net/http#HandleFunc). This function takes a function as a parameter. The parameter is named "handler".
* The [sort](https://pkg.go.dev/sort) package includes a number of examples of functions that take functions as parameters too. They also use short, evocative names like "less" and "cmp".
* The [strings.Map](https://pkg.go.dev/strings#Map) function names a parameter function "mapping".
* The [Effective Go section on names](https://go.dev/doc/effective_go#names) puts emphasis on keeping names short and concise.

I interpret this to mean that there is no Go convention around using a parameter name to hint at its type. We certainly don't do this for other parameter types; a function definition of `func validate(nameString string, uppercaseBool bool)` would not be following convention. To me, a function that's also a function parameter should have a name that implies action. Hence "started" feels like a name that suggests action was taken, since it uses the past conjugation of the verb "to start".

I can rewrite the comment documentation to more clearly use the parameter name as a value and not as a verb, but I'm pretty sure that `started` is following about as close of a convention as I can find.